src/sinks/statsd.rs

```diff
@@ -448,21 +448,17 @@ mod test {
                 },
             }),
         ];
-        let (tx, rx) = mpsc::channel(1);
+        let (mut tx, rx) = mpsc::channel(1);
 
         let socket = UdpSocket::bind(addr).await.unwrap();
         tokio::spawn(async move {
-            let stream = UdpFramed::new(socket, BytesCodec::new())
+            let mut stream = UdpFramed::new(socket, BytesCodec::new())
                 .map_err(|error| error!(message = "Error reading line.", %error))
                 .map_ok(|(bytes, _addr)| bytes.freeze());
 
-            let mut tx = tx
-                .sink_compat()
-                .sink_map_err(|error| error!(message = "Error sending event.", %error));
-
-            stream.forward(&mut tx).await.unwrap();
-
-            tx.flush().await.unwrap();
+            while let Some(Ok(item)) = stream.next().await {
+                tx.send(item).await.unwrap();
+            }
```

Comment:
This looks like `forward`/`send_all` :thinking:

Reply:
Yes, this is because `tokio::sync::mpsc::Sender` does not implement `Sink` itself. In most places like this, we use `Pipeline`, which does, but in this case, we are sending `Bytes`, not events. It's a little bit weird but seemed like the simplest thing to do.